ast/modify_test.go

```diff
@@ -119,3 +113,29 @@ func TestModify(t *testing.T) {
 		}
 	}
 }
+
+func TestModifyMap(t *testing.T) {
+	// need to test map separately because deepequal can't compare a map
+	// with keys of the same underlying value (2:2, 2:2)
+	modified := Modify(
+		&MapLiteral{
```

Comment:
DeepEquals cannot, but [maps.Equal](https://pkg.go.dev/maps#Equal) can, no?

Reply:
== wouldn't be equal as the values (and keys) are pointers to integer

Reply:
Oh indeed.